cyclonedx/model/bom.py

```diff
@@ -288,19 +291,19 @@ def components(self) -> "SortedSet[Component]":
     def components(self, components: Iterable[Component]) -> None:
         self._components = SortedSet(components)
 
-    def get_component_by_purl(self, purl: Optional[str]) -> Optional[Component]:
+    def get_component_by_purl(self, purl: Optional["PackageURL"]) -> Optional[Component]:
         """
         Get a Component already in the Bom by its PURL
 
         Args:
              purl:
-                Package URL as a `str` to look and find `Component`
+                An instance of `packageurl.PackageURL` to look and find `Component`.
 
         Returns:
             `Component` or `None`
         """
         if purl:
-            found = list(filter(lambda x: x.purl == purl, self.components))
+            found = [x for x in self.components if x.purl == purl]
```

Comment:
why changed?

Reply:
as described, list comprehensions are more performant than using list + filter + lambda. I timed it and it is roughly 2x faster.